test/migrations/014_add_constraint.js

```diff
@@ -1,7 +1,40 @@
 exports.constraint = 'chck_nmbr';
 
+/**
+ * @param pgm {import('node-pg-migrate').MigrationBuilder}
+ */
 exports.up = (pgm) => {
   pgm.addConstraint('t1', exports.constraint, {
     check: 'nmbr < 30',
+    comment: 'nmbr must be less than 30',
   });
+
+  // Test for issue #939
+  pgm.createSchema('payroll_reports');
+  pgm.createTable(
+    { schema: 'payroll_reports', name: 'upload_headers' },
+    {
+      id: { type: 'serial', primaryKey: true },
+      paycode_type: 'text',
+      aggregate_paycode_type: 'text',
+      meta_field_type: 'text',
+      is_remark: 'boolean',
+    }
+  );
+  pgm.addConstraint(
```

Comment:
do these tests just make sure that the migrations themselves don't error? or is there some notion of checking that the resulting schema comes out how was intended? I see that the integration tests invoked in the `postgres-test.yml` action runs these migrations, but a little surprised there's no assertions/expectations in the tests themselves.

Reply:
PostgreSQL itself would fail the migration. Look here for an example of a previous commit that failed: https://github.com/salsita/node-pg-migrate/actions/runs/8897624981/job/24432928126?pr=1139#step:8:321

<details>
<summary>Example failed action</summary>

```sql
### MIGRATION 014_add_constraint (UP) ###
ALTER TABLE "t1"
  ADD CONSTRAINT "chck_nmbr" CHECK (nmbr < 30);
COMMENT ON CONSTRAINT "chck_nmbr" ON "t1" IS $pga$nmbr must be less than 30$pga$;
CREATE SCHEMA "payroll_reports";
CREATE TABLE "payroll_reports"."upload_headers" (
  "id" serial PRIMARY KEY,
  "paycode_type" text,
  "aggregate_paycode_type" text,
  "meta_field_type" text,
  "is_remark" boolean
);
ALTER TABLE "payroll_reports"."upload_headers"
  ADD CONSTRAINT "chk_only_one_header_type" CHECK ((       CASE WHEN paycode_type IS NOT NULL THEN 1 ELSE 0 END +       CASE WHEN aggregate_paycode_type IS NOT NULL THEN 1 ELSE 0 END +       CASE WHEN meta_field_type IS NOT NULL THEN 1 ELSE 0 END +       CASE WHEN is_remark THEN 1 ELSE 0 END      ) <= 1);
COMMENT ON CONSTRAINT "chk_only_one_header_type" ON "upload_headers" IS $pga$If no type present/truthy, it's ignored; cannot have more than one active at once$pga$;
INSERT INTO "public"."pgmigrations" (name, run_on) VALUES ('014_add_constraint', NOW());


Error executing:
ALTER TABLE "payroll_reports"."upload_headers"
  ADD CONSTRAINT "chk_only_one_header_type" CHECK ((       CASE WHEN paycode_type IS NOT NULL THEN 1 ELSE 0 END +       CASE WHEN aggregate_paycode_type IS NOT NULL THEN 1 ELSE 0 END +       CASE WHEN meta_field_type IS NOT NULL THEN 1 ELSE 0 END +       CASE WHEN is_remark THEN 1 ELSE 0 END      ) <= 1);
COMMENT ON CONSTRAINT "chk_only_one_header_type" ON "upload_headers" IS $pga$If no type present/truthy, it's ignored; cannot have more than one active at once$pga$;
error: relation "upload_headers" does not exist

> Rolling back attempted migration ...
```

</details>

The integration tests are running against real PostgreSQL docker containers in all currently PostgreSQL supported versions => https://www.postgresql.org/
![image](https://github.com/salsita/node-pg-migrate/assets/7195563/a291f474-d421-4e27-9e30-65b0dc41ba26)